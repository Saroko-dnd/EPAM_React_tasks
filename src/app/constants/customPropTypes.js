import { createPropType } from 'react-custom-proptypes';

const articlePropType = createPropType(prop => (
    typeof prop === 'object' &&
        typeof prop.author === 'string' &&
        typeof prop.title === 'string' &&
        typeof prop.description === 'string' &&
        typeof prop.url === 'string' &&
        typeof prop.urlToImage === 'string' &&
        typeof prop.publishedAt === 'string' &&
        typeof prop.source === 'object' &&
        typeof prop.source.name === 'string'
), 'Must be news article');

export default { article: articlePropType };
