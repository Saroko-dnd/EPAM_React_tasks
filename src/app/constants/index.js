const newsAppConfig = {
    apiToken: 'e99a027c252b468ab4c0d3d8872f4cfc',
    apiBaseLink: 'https://newsapi.org/v2/',

    navigation: {
        homePage: { destination: '/', title: 'Top news', id: 'nav-link-1' },
        firstPage: {
            destination: '/first',
            title: 'Ultra news',
            id: 'nav-link-2',
        },
        secondPage: {
            destination: '/second',
            title: 'Super news',
            id: 'nav-link-3',
        },
        thirdPage: {
            destination: '/third',
            title: 'Mega news',
            id: 'nav-link-4',
        },
    },

    colors: {
        green: '#009900',
    },
};

export default newsAppConfig;
