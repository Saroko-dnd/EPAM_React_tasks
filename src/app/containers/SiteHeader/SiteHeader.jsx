import React from 'react';
import PropTypes from 'prop-types';

import Header from '../../components/Header';

const SiteHeader = ({ navLinksToggled, toggleNavLinks }) => (
    <header>
        <Header
            navLinksToggled={navLinksToggled}
            toggleNavLinks={toggleNavLinks}
        />
    </header>
);

SiteHeader.propTypes = {
    toggleNavLinks: PropTypes.func.isRequired,
    navLinksToggled: PropTypes.bool.isRequired,
};

export default SiteHeader;
