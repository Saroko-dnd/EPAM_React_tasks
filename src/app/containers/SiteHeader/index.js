import { connect } from 'react-redux';

import SiteHeader from './SiteHeader';
import actions from '../../actions';

const mapStateToProps = state => ({
    navLinksToggled: state.navLinksToggled,
});

const mapDispatchToProps = dispatch => ({
    toggleNavLinks: (toggle) => {
        dispatch(actions.toggleNavLinks(toggle));
    },
});

const mergeProps = (stateProps, dispatchProps) => ({
    ...stateProps,
    ...dispatchProps,
});

const options = { pure: false };

export default connect(
    mapStateToProps,
    mapDispatchToProps,
    mergeProps,
    options,
)(SiteHeader);
