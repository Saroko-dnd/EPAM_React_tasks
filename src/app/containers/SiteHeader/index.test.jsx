import SiteHeader from './SiteHeader';
import { actionTypes } from '../../constants';
import { __GetDependency__ } from './index';

describe('SiteHeader container component', () => {
    describe('rendering and behavior', () => {
        const fakeToogle = jest.fn();
        const menuIsOpened = false;
        const siteHeaderWrapper = shallow(<SiteHeader
            navMenuIsOpened={menuIsOpened}
            toggleNavLinks={fakeToogle}
        />);

        it('should render SiteHeader correctly', () => {
            expect(siteHeaderWrapper).toMatchSnapshot();
        });
    });

    describe('mapStateToProps', () => {
        const fakeState = {
            navMenuIsOpened: false,
        };
        const mapStateToProps = __GetDependency__('mapStateToProps');
        const mappingResult = mapStateToProps(fakeState);

        it('should map property navMenuIsOpened from application state to SiteHeader property navMenuIsOpened', () => {
            expect(mappingResult.navMenuIsOpened).toEqual(fakeState.navMenuIsOpened);
        });
    });

    describe('mapDispatchToProps', () => {
        const fakeDispatch = jest.fn();
        const isMenuOpened = true;
        const mapDispatchToProps = __GetDependency__('mapDispatchToProps');

        it('toggleNavLinks(boolean) should open/close navigation menu', () => {
            mapDispatchToProps(fakeDispatch).toggleNavLinks(isMenuOpened);
            expect(fakeDispatch).toBeCalledWith({
                type: actionTypes.TOGGLE_NAV_MENU,
                payload: isMenuOpened,
            });
        });
    });

    describe('mergeProps', () => {
        const firstTestProps = {
            width: 200,
            height: 150,
        };
        const secondTestProps = {
            name: 'rat',
            id: '76grass89',
        };

        const mergeProps = __GetDependency__('mergeProps');
        const mergedProps = mergeProps(firstTestProps, secondTestProps);

        it('should merge all properties in one object', () => {
            expect(mergedProps).toEqual({
                ...firstTestProps,
                ...secondTestProps,
            });
        });
    });
});
