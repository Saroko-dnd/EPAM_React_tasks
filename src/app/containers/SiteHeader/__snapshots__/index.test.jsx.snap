// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SiteHeader container component rendering and behavior should render SiteHeader correctly 1`] = `
<header>
  <Header
    navMenuIsOpened={false}
    toggleNavLinks={[MockFunction]}
  />
</header>
`;
